src/store/checkout-store.js

```diff
@@ -35,6 +36,11 @@ store.registerHandler('NEW_CHECKOUT', data => {
     if (StudentStore.hasOverdueItem(data.studentId)) {
         throw new Error('Student has overdue item');
     }
+    data.itemAddresses.forEach(itemAddress => {
+        if (ItemStore.getItemByAddress(itemAddress).status !== 'AVAILABLE') {
+            throw new Error('An item in the cart is not available for checkout.');
```

Comment:
So if this error gets thrown, is the entire checkout discarded, or just the unavailable item?